src/open_inwoner/accounts/views/inbox.py

```diff
@@ -126,24 +142,31 @@ def get_form_kwargs(self):
     def form_valid(self, form):
         object = form.save()
 
-        # build redirect url based on form hidden data
-        url = furl(self.request.path).add({"with": form.data["receiver"]}).url
+        url = reverse(
+            "accounts:inbox",
+            kwargs={self.slug_field: form.cleaned_data["receiver"].uuid},
+        )
 
         self.log_addition(object, _("message was created"))
         return HttpResponseRedirect(f"{url}#messages-last")
 
     def get(self, request, *args, **kwargs):
-        """Mark all messages as seen for the receiver"""
         context = self.get_context_data()
 
-        # Redirect to the end of page.
-        # Redirecting to a hash doesn't work, so we need to change the url.
-        # Alter URL with redirected query param in order to go to the last message (#messages-last).
+        """
```

Comment:
By removing this part here it seems that the `message.seen`  is never updated to True. That's why Mijn berichten (navigation bar) is not updated with the number of seen=False messages.
![image](https://user-images.githubusercontent.com/63058071/208662089-dcb0d109-e34f-4fe7-ab2b-e23a42af8f07.png)

Reply:
Good catch. I'll need to fix it and see about the tests.

Reply:
Tested again the code and works as expected. Maybe something was wrong with the browsers.